### DISCUSSION TITLE
Question about videos

### DISCUSSION THREAD

**Discussion by switchmaxfx** [TheZoraiz/ascii-image-converter#20] [Ideas]
I have this on my raspberry pi serving as an art installation and I wanted to ask if you have any plans on getting this to support video? Or know of another similar program that does the same thing with video? Im using Xterm for my terminal.

**Comment by TheZoraiz** [comment]
I don't plan to add video support, and it'd come in the form of a different tool if I did lol. That said, [junguler](https://github.com/junguler) has done an amazing job of documenting the varying usecases of this tool, which includes video convesions:
https://github.com/junguler/_image-manipulation/tree/main/Ascii-image-converter

This might interest you as well:
https://github.com/joelibaceta/video-to-ascii

**Reply by switchmaxfx** [reply]
> I don't plan to add video support, and it'd come in the form of a different tool if I did lol. That said, [junguler](https://github.com/junguler) has done an amazing job of documenting the varying usecases of this tool, which includes video convesions: https://github.com/junguler/_image-manipulation/tree/main/Ascii-image-converter
> 
> This might interest you as well: https://github.com/joelibaceta/video-to-ascii

Thanks for the info! Im mainly trying to get gifs playing. Should have added that in the first post instead of videos.

**Comment by switchmaxfx** [comment]
Is it possible to run the command by selecting a random file in a directory? Im trying to get a script setup to select a random file in my pictures folders and at certain times.

**Reply by TheZoraiz** [reply]
Here's a simple bash script to do this in linux:

````
ls | sort -R | while read file; do
	ascii-image-converter $file
done
````

Move this to the directory you want and execute :)

**Reply by switchmaxfx** [reply]
> Here's a simple bash script to do this in linux:
> 
> ```
> ls | sort -R | while read file; do
> 	ascii-image-converter $file
> done
> ```
> 
> Move this to the directory you want and execute :)

This works but some files post an error.

"unable to open file: open FILE.png: no such file or directory

**Comment by switchmaxfx** [comment]
Edit, wait i see some of my files have spaces in between and I see it will display the first bit then another like 

unable to open file: open FILE : no such file or directory

unable to open file: open #1.png: no such file or directory

the file would be "FILE #1.png"

EDIT 
don't worry I got it working. i replaced the $file with " *.jpg and *.png" 

Another issue was even though its a random tag it still prints them out in order from top to bottom in the folder. i changed the "sort -R" to "shuf -n 1" but keeps printing in the same sequence. Also do you know how I would time this? as in running the script in intervals of say an hour or 30 mins because right now its spitting them all out at once. Ty for the help with this.

**Reply by TheZoraiz** [reply]
Ah, I see. ascii-image-converter was designed to process multiple files, so it reads `FILE #1.png` as two different files because of the space. Just wrap the `$file` in the above script in inverted commas like so:

```sh
ls | sort -R | while read file; do
	ascii-image-converter "$file"
done
```

Also, if you want to just convert 1 random image from a directory, this would do the trick:

```sh
ls | sort -R | tail -n 1 | while read file; do
	ascii-image-converter "$file"
done
```

**Reply by TheZoraiz** [reply]
> Another issue was even though its a random tag it still prints them out in order from top to bottom in the folder. i changed the "sort -R" to "shuf -n 1" but keeps printing in the same sequence. Also do you know how I would time this? as in running the script in intervals of say an hour or 30 mins because right now its spitting them all out at once. Ty for the help with this.

Putting `sleep 1800` in the loop would do the trick

**Reply by TheZoraiz** [reply]
> don't worry I got it working. i replaced the $file with " *.jpg and *.png"

Don't do that. That just causes it to convert all files in every iteration of the loop. Try the modified scripts I wrote above.

**Reply by TheZoraiz** [reply]
In essence, this would be the script to print a random image from a directory every 30 minutes (make sure there are only images in the directory though): 

```bash
ls | sort -R | while read file; do
	ascii-image-converter "$file"
	sleep 1800
	clear
done
```

**Reply by switchmaxfx** [reply]
> In essence, this would be the script to print a random image from a directory every 30 minutes (make sure there are only images in the directory though):
> 
> ```shell
> ls | sort -R | while read file; do
> 	ascii-image-converter "$file"
> 	sleep 1800
> 	clear
> done
> ```

This does the trick but it doesn't continue to run it after the first image is printed. it just clears then doesn't run again. 

EDIT 
IT WORKS!
i ended up typing out some tags wrong in the command but now it runs perfectly. I'm going to make a project on hackster.io and link you. thank you again!
